internal/logs/log_container.go

```diff
@@ -0,0 +1,94 @@
+package logs
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+)
+
+const (
+	elapsedTime = 100 * time.Millisecond
+)
+
+type LogEntry struct {
+	Time     time.Time
+	data     []byte
+	workload string
+}
+
+func NewLogEntry(data []byte, workload string) *LogEntry {
+	return &LogEntry{
+		data:     data,
+		Time:     time.Now(),
+		workload: workload,
+	}
+}
+
+func (l *LogEntry) Size() int {
+	return len(l.data)
+}
+
+func (l *LogEntry) String() string {
+	if l.data == nil {
+		return ""
+	}
+	return string(l.data)
+}
+
+func (l *LogEntry) GetWorkload() string {
+	return l.workload
+}
+
+type FifoLog struct {
+	maxlen      int
+	data        []*LogEntry
+	lock        sync.Mutex
+	currentSize int
+}
+
+// NewFifoLog returns a new FifoLog struct
+func NewFifoLog(maxSize int) *FifoLog {
+	return &FifoLog{
+		data:   []*LogEntry{},
+		maxlen: maxSize,
+	}
+}
+
+// CurrentSize returns the buffer size.
+func (f *FifoLog) CurrentSize() int {
+	f.lock.Lock()
```

Comment:
What is the reason for this lock?

Reply:
Write is called from another goroutine(WorkloadStarted) and Readline from target (Another goroutine) Current size is moved increased/decreased on each function, so lock is needed.